### ISSUE TITLE
skopeo inspect: a way how to avoid fetching all tags from repository

### ISSUE THREAD

**Issue by MartinBasti** [containers/skopeo#785]
Hello,

in our use cases we need to get only information about image, like labels, but we don't need all `repoTags`. We have repositories with many tags (1700+) and it is resource consuming to get all tags from registry and just drop that.

Could you please provide a way how to avoid fetching repotags?

Thank you

**Comment by lcarva** [comment]
This would be very useful if we want to use skopeo to simply determine the digest of the manfiest. Currently, I don't think there's a way to retrieve this information without also retrieving all the tags in the repository (which is very inefficient for repositories with many tags).

**Comment by rhatdan** [comment]
Since this can not be done based on the registry protocol, I think this issue should be close.  Reopen if I am mistaken.

**Comment by mtrmac** [comment]
Filtering can’t be done, not getting the data at all definitely can be done.

**Comment by rittneje** [comment]
+1 on this. Since the `RepoTags` have nothing to do with the specific image that is being inspected (#1208), it's actually very confusing that it's in the output at all. I think it should be removed in a future major version of skopeo. Plus that list will continue to grow without bound over time and make things slow. (For reference the ubuntu registry currently has over 400 tags and counting.)

**Comment by airadier** [comment]
Some registries like Public ECR don't seem to support listing tags, so Skopeo inspect will fail:
```
skopeo inspect docker://public.ecr.aws/jg/hello:latest
FATA[0004] Error determining repository tags: Error fetching tags list: invalid status code from registry 404 (Not Found)
```

Using --raw will work instead.

**Comment by gsaslis** [comment]
+1 @rittneje 

I also found RepoTags *pretty* confusing when inspecting a specific **image**. Agree on removing it altogether because it belongs to a different entity: the Repository, not the Image. 

i.e. this seems like it is both slow and out-of-place.

**Comment by mtrmac** [comment]
> Some registries like Public ECR don't seem to support listing tags

Thanks, let’s track that separately in #1230.

**Comment by lsm5** [comment]
> I'm using `skopeo inspect --config <pullspec>` if I need information about labels. It's not fetching tags


@vrothberg @mtrmac is this sufficient to close this, perhaps accompanied with a manpage update?

**Comment by ktdreyer** [comment]
When I run `skopeo inspect docker://docker.io/library/python:3.9`, I expect to only look at the single "3.9" tag. It works for repos with a few tags, and silently hangs for repos with thousands of tags. I'm already giving skopeo a single tag - why query pages and pages of data from the server just to throw it away? This can trigger rate-limiting on the server, etc.

It's great that `--config` shortcuts loading all tags, but I expect new users won't discover this easily. They'll just think skopeo is broken.

**Comment by mtrmac** [comment]
> > I'm using `skopeo inspect --config <pullspec>` if I need information about labels. It's not fetching tags
> 
> @vrothberg @mtrmac is this sufficient to close this, perhaps accompanied with a manpage update?

No; it doesn’t allow using any of the other modes.

**Comment by mtrmac** [comment]
AFAICS there’s no difference.

Right now, the list of tags in `RepoTags` is a list of all tags in the repository, whether or not the CLI input specified a tag (`:3.9`), a digest `@sha256:…`), or neither (defaulting to `:latest`); that part of the CLI input has never affected the `RepoTags` value, but it did affect all other parts of the output (by choosing which image is inspected).

The proposed `--no-tags` option likewise disables collecting and printing the `RepoTags` value, and doesn’t affect any other part of the input.

Is there anything I’m missing?

**Comment by arewm** [comment]
Can this issue be closed now that #1480 is merged and `1.5.1` is released?

**Comment by rhatdan** [comment]
SGTM
